crates/tools/src/main.rs

```diff
@@ -29,31 +29,39 @@ fn main() -> Result<()> {
 }
 
 fn install_code_extension() -> Result<()> {
-    run("cargo install --path crates/ra_lsp_server --force", ".")?;
-    if cfg!(windows) {
-        run(r"cmd.exe /c npm.cmd ci", "./editors/code")?;
-        run(r"cmd.exe /c npm.cmd run package", "./editors/code")?;
-    } else {
-        run(r"npm ci", "./editors/code")?;
-        run(r"npm run package", "./editors/code")?;
-    }
+   run("cargo", "install --path crates/ra_lsp_server --force", ".")?;
+   if cfg!(windows) {
+       run(r"cmd.exe", "/c npm.cmd ci", "./editors/code")?;
+       run(r"cmd.exe", "/c npm.cmd run package", "./editors/code")?;
+   } else {
+       run(r"npm", "ci", "./editors/code")?;
+       run(r"npm", "run package", "./editors/code")?;
+   }
     if cfg!(windows) {
         run(
-            r"cmd.exe /c code.cmd --install-extension ./ra-lsp-0.0.1.vsix --force",
+            r"cmd.exe", r"/c code.cmd --install-extension ./ra-lsp-0.0.1.vsix --force",
             "./editors/code",
         )?;
+    } else if cfg!(target_os = "macos") {
+        let vscode_exec = get_vscode_exec()?;
```

Comment:
This is a "hacking goes here"-style code, so perhaps we could use `set_env` to append `/Applications/Visual Studio Code.app/Contents/Resources/app/bin/` to `PATH`, and leave most of the code as is?

Reply:
Yes, using `PATH` would be better, than getting executable. Maybe move `install_code_extension` and `verify_installed_extensions` to separate binary where in first step setuped environment and then install lsp, build, install and verify extension?

Reply:
I think we can just unconditionally append to the PATH on macos without splitting anything?

Reply:
Added new module where append vscode path. Before calling `install_code_extension` add setuping environment.